shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -21,11 +23,44 @@ use tracing::{debug, trace, warn};
 /// <https://docs.substrate.io/reference/scale-codec/#fn-1>
 pub const MAX_SEGMENT_PADDING: usize = 3;
 
-/// The maximum object length this module can handle.
+/// The minimum size of the segment header.
+//
+// V0 variant + segment index + segment commitment + prev segment header hash +
+// last archived block: number + progress variant (no data in the variant).
+pub const MIN_SEGMENT_HEADER_SIZE: usize =
+    1 + 8 + SegmentCommitment::SIZE + Blake3Hash::SIZE + 4 + 1;
+
+/// The maximum size of the segment header.
+//
+// All of the above, plus the partial byte count for the partial progress variant.
+pub const MAX_SEGMENT_HEADER_SIZE: usize = MIN_SEGMENT_HEADER_SIZE + 4;
+
+/// The maximum object length the implementation in this module can reliably handle.
 ///
-/// Currently objects are limited by the largest block size in any domain, which is 5 MB.
-/// But this implementation supports the maximum length of the 4 byte scale encoding.
-pub const MAX_SUPPORTED_OBJECT_LENGTH: usize = 1024 * 1024 * 1024 - 1;
+/// Currently objects are limited by the largest block size in any domain, which is 5 MB. But this
```

Comment:
not sure if `domain` is the right word here since it we have a different context where we use it for domains